File: src/main/java/fr/klemek/englishparser/model/dict/Noun.java
package fr.klemek.englishparser.model.dict;

import fr.klemek.englishparser.model.DatabaseObject;
import fr.klemek.englishparser.utils.DatabaseManager;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.List;
import java.util.Objects;

@Entity
@Table(name = "dict_noun")
public class Noun extends WordObject {

    @Id
    @Column(name = "plural")
    private String plural;
    @Column(name = "male")
    private boolean male;
    @Column(name = "female")
    private String female;
    @Column(name = "female_plural")
    private String femalePlural;

    public Noun() {
    }

    public Noun(String base, String plural) {
        super(base);
        this.plural = plural;
    }

    public Noun(String base, String plural, String female, String femalePlural) {
        this(base, plural);
        this.setFemale(female, femalePlural);
    }

    public void setFemale(String female, String femalePlural) {
        this.male = female != null;
        this.female = female;
        this.femalePlural = femalePlural;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Noun noun = (Noun) o;
        return male == noun.male &&
                Objects.equals(plural, noun.plural) &&
                Objects.equals(female, noun.female) &&
                Objects.equals(femalePlural, noun.femalePlural);
    }

    @Override
    public int hashCode() {

        return Objects.hash(super.hashCode(), plural, male, female, femalePlural);
    }

    public static List<Noun> getAll() {
        return DatabaseObject.getAll(Noun.class);
    }

    @Override
    public String toString() {
        return "Noun{" +
                "base='" + getBase() + '\'' +
                ", plural='" + plural + '\'' +
                ", male=" + male +
                ", female='" + female + '\'' +
                ", femalePlural='" + femalePlural + '\'' +
                '}';
    }

    public static Noun getByFemale(String word) {
        return DatabaseManager.getFirstFromSessionQuery("FROM Noun WHERE female = ?0", word);
    }

    public static Noun getByWord(String word) {
        return DatabaseManager.getFirstFromSessionQuery("FROM Noun WHERE base = ?0 OR plural = ?0", word);
    }

    public static List<Noun> searchByWord(String word) {
        return DatabaseManager.getRowsFromSessionQuery("FROM Noun WHERE base = ?0 OR plural = ?0 OR (male = 1 AND (female = ?0 OR female_plural = ?0))", word);
    }
}

File: EnglishParser.Model/Noun.cs
using System;

namespace EnglishParser.Model
{
    public class Noun : WordObject
    {
        public Noun()
        {
        }

        public Noun(string @base, string plural) : base(@base)
        {
            Plural = plural ?? throw new ArgumentNullException(nameof(plural));
        }

        public Noun(string @base, string plural, string female, string femalePlural) : this(@base, plural)
        {
            SetFemale(female, femalePlural);
        }

        public string Plural { get; set; }
        public string Female { get; set; }
        public string FemalePlural { get; set; }

        public bool Proper => char.IsUpper(Base[0]);

        public void SetFemale(string female, string femalePlural)
        {
            Female = female;
            FemalePlural = femalePlural;
        }

        protected bool Equals(Noun other)
        {
            return base.Equals(other) && Plural == other.Plural && Female == other.Female && FemalePlural == other.FemalePlural;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Noun) obj);
        }

        public override int GetHashCode()
        {
            unchecked
            {
                int hashCode = base.GetHashCode();
                hashCode = (hashCode * 397) ^ (Plural != null ? Plural.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (Female != null ? Female.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (FemalePlural != null ? FemalePlural.GetHashCode() : 0);
                return hashCode;
            }
        }
    }
}